openedx/core/djangoapps/content/course_overviews/tasks.py

```diff
@@ -0,0 +1,54 @@
+import logging
+
+from celery import task
+from celery_utils.logged_task import LoggedTask
+from celery_utils.persist_on_failure import PersistOnFailureTask
+from django.conf import settings
+
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+log = logging.getLogger(__name__)
+
+
+DEFAULT_CHUNK_SIZE = 50
+
+
+def chunks(sequence, chunk_size):
+    return (sequence[index: index + chunk_size] for index in xrange(0, len(sequence), chunk_size))
+
+
+class _BaseTask(PersistOnFailureTask, LoggedTask):  # pylint: disable=abstract-method
+    """
+    Include persistence features, as well as logging of task invocation.
+    """
+    abstract = True
+
+
+def generate_course_overview(course_ids, all_courses=False, force_update=False, chunk_size=DEFAULT_CHUNK_SIZE):
+    """
+    Invokes asynchronous celery tasks to load course overviews for all courses specified in course_ids.
+    If all_courses is True, loads a course overview for every course in the modulestore.
```

Comment:
The docstring seems superfluous, given the code is self-explanatory.  Consider renaming this method so the 1st sentence is captured in the method's name.  For example, `async_update_courses`.  Also, it may cause unnecessary confusion to have 2 different names within this module: `generate_course_overview` versus `update_select_courses` when the latter is just subtasks of the former.

Reply:
👍